app/routes/circulars/circulars.server.ts

```diff
@@ -172,3 +173,19 @@ export async function put(subject: string, body: string, request: Request) {
     submitter: formatAuthor(user),
   })
 }
+
+export async function circularRedirect(query: string | undefined) {
+  if (!query) return
+  const validCircularSearchStyles =
+    /^(?:\s*GCN\s*)?(?:\s*CIRCULAR\s*)?((\s*-?\d+(?:\.\d)?)\s*)$/i
+  const circularID = validCircularSearchStyles.exec(query)?.[1]
+  if (circularID) {
+    const db = await tables()
+    const result = await db.circulars.get({
+      circularId: Number(circularID),
+    })
+    if (!result) return
+    const circularURL = `/circulars/${circularID}`
```

Comment:
Very slightly terser:

```suggestion
  const circularId = parseFloat(validCircularSearchStyles.exec(query)?.[1])
  if (!isNaN(circularID)) {
    const db = await tables()
    const result = await db.circulars.get({ circularId })
    const circularURL = `/circulars/${circularId}`
```

Reply:
line 187 of the commit is handling the case where the circular Id doesn't exist, I believe the omission of that will throw an error if you try to search a circular Id that doesn't exist. I was able to remove the first line of the function (the conditional where query is undefined) by tweaking your line 181 to be 
`const circularID = parseFloat(validCircularSearchStyles.exec(query)?.[1]) ?? NaN`
and using your suggested conditional on line 182

Reply:
`parseFloat` returns NaN if its argument is null, undefined, or an invalid string --- so there is no need for the `?? NaN`.

Reply:
Ah, but you're right, you do need the `if (!result) return` line.

Reply:
ah, okay. I am having an issue where parseFloat on that line is causing the pre-commit to fail because 

> Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
>   Type 'undefined' is not assignable to type 'string'.

though this issue is persisting after altering index.tsx to only call circularRedirect if the query is not undefined and changing the query argument to only be accepted as a string. Looking online, this might be an issue with my intellisense on VSCode